server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -322,16 +322,22 @@ public void updateCachedBlockIds(String appId, int shuffleId, int partitionId, S
         size += spb.getSize();
       }
     }
-    shuffleTaskInfo.addPartitionDataSize(
+    long partitionSize = shuffleTaskInfo.addPartitionDataSize(
         shuffleId,
         partitionId,
         size
     );
+    if (shuffleBufferManager.isHugePartition(partitionSize)) {
```

Comment:
> I think this's the better place to update related metrics here.

Sounds good. Let me take a look tomorrow 


> Once shuffleBufferManager detects a huge partition, it cloud increase the app_num_with_huge_partition metrics if the appId is not already added to hasSeenAppIdSet.

Actually I don’t hope we maintain the  
hasSeenAppIdSet data structure in shuffleBufferManager. This info bound to shuffleTaskInfo will be better ? WDYT

Reply:
> One appId could had multiple shuffleTaskInfos, right?

One appId only has one shuffleTaskInfo.

> Once shuffleBufferManager detects a huge partition, it cloud increase the app_num_with_huge_partition metrics if the appId is not already added to hasSeenAppIdSet.

How to maintain `hasSeenAppIdSet`'s lifecycle? Should one element of one appId in  `hasSeenAppIdSet` keep consistent with its `shuffleTaskInfo` lifecycle? If yes, it also will be removed due to heartbeat timeout. Otherwise, we have to maintain `hasSeenAppIdSet` lifecycle independently, this is too complex and unreasonable.

Reply:
> My fault. The class name is kind of misleading, I used to think this class corresponding to one shuffle process/task, mapping to one shuffleId in spark app.

Yes, this class name is a litter bit misleading. Let's rename it in another PR? Do you want to push this?

> But that may be a little bit of complex..

+1